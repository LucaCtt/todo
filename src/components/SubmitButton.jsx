import React, { memo } from "react";
import { View, StyleSheet } from "react-native";
import { Button, Spinner } from "@ui-kitten/components";

const LoadingIndicator = ({ ...props }) => (
  <View {...props} accessibilityLabel="loading">
    <Spinner />
  </View>
);

/**
 * A button that supports showing a loading animation, meant to be used to submit forms.
 *
 * @property {boolean} isLoading - Whether the loading animation should be shown. Defaults to *false*.
 */
const SubmitButton = ({ isLoading = false, style, children, ...props }) => (
  <Button
    {...props}
    accessoryLeft={isLoading ? LoadingIndicator : null}
    style={[styles.button, style]}
  >
    {isLoading ? null : children}
  </Button>
);

const styles = StyleSheet.create({
  button: {
    marginVertical: 10,
  },
});

export default memo(SubmitButton);
