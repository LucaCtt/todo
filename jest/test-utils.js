import React from "react";
import { render } from "@testing-library/react-native";
import * as eva from "@eva-design/eva";
import { EvaIconsPack } from "@ui-kitten/eva-icons";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";

const TestProvider = ({ children }) => (
  <>
    <IconRegistry icons={EvaIconsPack} />
    <ApplicationProvider mapping={eva.mapping} theme={eva.light}>
      {children}
    </ApplicationProvider>
  </>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: TestProvider, ...options });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
