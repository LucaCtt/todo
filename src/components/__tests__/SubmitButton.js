import React from "react";
import { render } from "test-utils";

import SubmitButton from "../SubmitButton";

describe("<SubmitButton />", () => {
  test("has 1 child", () => {
    const { baseElement } = render(<SubmitButton isLoading />);
    expect(baseElement).toMatchSnapshot();
  });
});
