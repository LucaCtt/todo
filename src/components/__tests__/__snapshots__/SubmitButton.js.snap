// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> has 1 child 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={1}
        appearance="filled"
        size="medium"
        status="primary"
        style={
          Array [
            undefined,
            Array [
              Object {
                "backgroundColor": "#3366FF",
                "borderColor": "#3366FF",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "minWidth": 40,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              },
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginVertical": 10,
                },
                undefined,
              ],
            ],
          ]
        }
        testID="submit-button"
      >
        <View
          style={
            Object {
              "height": 20,
              "marginHorizontal": 10,
              "tintColor": "#FFFFFF",
              "width": 20,
            }
          }
        >
          <View
            style={
              Size {
                "height": 24,
                "width": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    undefined,
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Size {
                          "height": 12,
                          "width": 24,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderColor": "#3366FF",
                            "borderRadius": 12,
                            "borderWidth": 3.4,
                            "height": 24,
                            "width": 24,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "bottom": 0,
                  "height": 24,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "top": 12,
                    },
                    Size {
                      "height": 12,
                      "width": 24,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "translateY": -12,
                        },
                        Object {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 24,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Size {
                          "height": 12,
                          "width": 24,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderColor": "#3366FF",
                            "borderRadius": 12,
                            "borderWidth": 3.4,
                            "height": 24,
                            "width": 24,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
